RecoLocalTracker/Phase2TrackerRecHits/plugins/VectorHitBuilderEDProducer.cc

```diff
@@ -0,0 +1,134 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoLocalTracker/Phase2TrackerRecHits/interface/VectorHitBuilderAlgorithm.h"
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class VectorHitBuilderEDProducer : public edm::stream::EDProducer<>
+{
+
+ public:
+
+  explicit VectorHitBuilderEDProducer(const edm::ParameterSet&);
+  virtual ~VectorHitBuilderEDProducer();
+  virtual void produce(edm::Event&, const edm::EventSetup&) override;
+  void setupAlgorithm(edm::ParameterSet const& conf);
+  void run(edm::Handle< edmNew::DetSetVector<Phase2TrackerCluster1D> > clusters,
+           edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersAcc, edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersRej,
+           VectorHitCollectionNew& outputAcc, VectorHitCollectionNew& outputRej);
+//  VectorHitBuilderAlgorithm * algo() const { return builderAlgo; };
+
+ private:
+  VectorHitBuilderAlgorithm* builderAlgo;
+  std::string offlinestubsTag;
+  unsigned int maxOfflinestubs;
+  std::string algoTag;
+  edm::EDGetTokenT< edmNew::DetSetVector<Phase2TrackerCluster1D> > clusterProducer;
+  bool readytobuild;
+
+};
+
+VectorHitBuilderEDProducer::VectorHitBuilderEDProducer(edm::ParameterSet const& conf)
+  : offlinestubsTag( conf.getParameter<std::string>( "offlinestubs" ) ),
+    maxOfflinestubs(conf.getParameter<int>( "maxVectorHits" )),
+    algoTag(conf.getParameter<std::string>( "Algorithm" )),
+    readytobuild(false)
+{
+
+  clusterProducer = consumes< edmNew::DetSetVector<Phase2TrackerCluster1D> >(edm::InputTag(conf.getParameter<std::string>("Clusters")));
+
+  produces< edmNew::DetSetVector< Phase2TrackerCluster1D > >( "ClustersAccepted" );
+  produces< edmNew::DetSetVector< Phase2TrackerCluster1D > >( "ClustersRejected" );
```

Comment:
Are these copies from the input clusters?  If yes, could we use refs instead of copies?